Extensions/classic_tags.js

```diff
@@ -225,163 +230,139 @@ XKit.extensions.classic_tags = new Object({
 					if (existing_count.length) {
 						existing_count.text(count);
 					} else {
-						anchor.find(".result_title").removeClass("no_count").after("<span class='count'>" + count + "</span>");
+						anchor.find(".result_title").after(`<span class="count">${count}</span>`);
 					}
 				});
 			});
 		} else {
-			$("#popover_search .result_link").each(function() {
-				var link = $(this);
-				var tag_name = link.attr("data-tag-result");
-				var count = self.tagcounts[tag_name];
+			self.tags.tags.forEach(tag => {
+				var count = self.tagcounts[tag.name];
+
 				if (!count || parseInt(count, 10) < self.max_posts_per_tag) {
-					fetch_count(tag_name);
+					fetch_count(tag.name);
 				}
 			});
 		}
 
-		var search = $("#search_query");
-		var new_label = "Search [new]";
-		if (self.preferences.show_new_notification.value && search.attr("placeholder") !== new_label) {
+		if (self.preferences.show_new_notification.value) {
 			$.when.apply($, new_post_count_promises).then(function() {
 				var any_new_posts = Array.prototype.some.call(arguments, function(count) { return !!count; });
-				if (any_new_posts) {
-					search.attr("placeholder", new_label);
-				}
+				self.search_input.attr("placeholder", `${self.placeholder}${(any_new_posts ? " [New]" : "")}`);
 			});
 		}
 
 		self.count_update_handle = setTimeout(self.update_tag_counts.bind(self, next_update * 1.5), next_update);
 	},
 
 	run: function() {
+		XKit.tools.init_css("classic_tags");
 
-		try {
+		var where = XKit.interface.where();
 
-			XKit.tools.init_css("classic_tags");
-			if ($("#dashboard_index").length === 0) {
-				if (document.location.href.indexOf('/tagged') === -1) {
-					return;
-				}
-			}
-
-			if (XKit.interface.where().tagged && !location.href.match(/before=[0-9]+/i)) {
-				XKit.extensions.classic_tags.update_tag_timestamp().then(function() {
-					XKit.extensions.classic_tags.show();
-				});
-			} else {
-				XKit.extensions.classic_tags.show();
-			}
-
-		} catch (e) {
+		if (!where.dashboard && !where.tagged) {
+			return;
+		}
 
-			console.error("Can't run Classic Tags:" + e.message);
+		XKit.css_map.getCssMap().then(() => {
+			try {
+				this.typeahead_dropdown = XKit.css_map.keyToCss("typeaheadDropdown");
+				this.tag_text = XKit.css_map.keyToCss("tagText");
 
-		}
+				if (where.tagged) {
+					this.update_tag_timestamp().then(() => this.show());
+				} else {
+					this.show();
+				}
+			} catch (e) {
+				console.error("Can't run Classic Tags:" + e.message);
+			}
+		});
```

Comment:
this `try` can be removed/unindented, and you can add `.catch` to the end of the `getCssMap()` promise to catch any errors in the `then` function, i.e.

```js
XKit.css_map.getCssMap().then(() => {
    // this.typeahead_dropdown = ...
    // etc.
}).catch(e => console.error("Can't run Classic Tags: " + e.message));